### PR TITLE
bartender: 4.2.21 -> 5.0.49

### PR DESCRIPTION
**Pull request by afh**
## Description of changes

#289900 has been merged ✅ ~~❗ The changes in this PR depend on #289900 as the `Bartender 5.dmg` contains an APFS filesystem which is currently [unsupported by native undmg](https://github.com/matthewbauer/undmg/blob/2d9c949fabf5aadd8e9470a65bef46ce36f6b76e/main.c#L65).~~

Changelog: https://www.macbartender.com/Bartender5/release_notes/

* Update to Bartender 5
* Improve derivation by using `${lib.versions.major finalAttrs.version}` in `src.name` and `src.url` and using `${finalAttrs.sourceRoot}` in `installPhase`.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/darwin/bartender/default.nix [hunk 1]
```diff
@@ -1,33 +1,33 @@
 { lib
 , stdenvNoCC
 , fetchurl
-, undmg
+, 7zz
```
Comment (afh) [inline pkgs/os-specific/darwin/bartender/default.nix:4]: ℹ️ #289900 adds a setup-hook to to unpack APFS DMGs using 7zz.

File: pkgs/os-specific/darwin/bartender/default.nix [hunk 2]
```diff
@@ -1,33 +1,33 @@
 { lib
 , stdenvNoCC
 , fetchurl
-, undmg
+, _7zz
```
Comment (afh) [inline pkgs/os-specific/darwin/bartender/default.nix:4]: ℹ️ #289900 adds a setup-hook to to unpack APFS DMGs using 7zz.

**Comment by afh** [comment]
Kindly requesting review and hoping for official approval from you, as the maintainer, @stepbrobd, now that support to unpack APFS DMGs has been merged with #289900.

**Comment by stepbrobd** [comment]
Result of `nixpkgs-review pr 289901` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bartender</li>
  </ul>
</details>

**Review by stepbrobd — APPROVED** [review]

**Comment by afh** [comment]
Thanks for merging, @SuperSandro, much appreciated 🥳

**Pull request merged**